llvm/utils/TableGen/GlobalISelEmitter.cpp

```diff
@@ -1901,7 +1901,12 @@ std::optional<CodeGenSubRegIndex *> GlobalISelEmitter::inferSubRegIndexForNode(
   return CGRegs.getSubRegIdx(SubRegInit->getDef());
 }
 
-Expected<RuleMatcher> GlobalISelEmitter::runOnPattern(const PatternToMatch &P) {
+std::optional<Expected<RuleMatcher>>
+GlobalISelEmitter::runOnPattern(const PatternToMatch &P) {
+  if (P.getISelShouldIgnore()) {
+    return {};
+  }
```

Comment:
You can just check this in `run` before calling `runOnPattern`, then you don't need to change `runOnPattern` signature at all.